s3/src/main/scala/geotrellis/spark/io/s3/package.scala

```diff
@@ -7,5 +7,14 @@ package object s3 {
   def makePath(chunks: String*) =
     chunks.filter(_.nonEmpty).mkString("/")
 
-  implicit class withSaveToS3Methods[K](rdd: RDD[(K, Array[Byte])]) extends SaveToS3Methods[K](rdd)
+  implicit class withSaveToS3Methods[K](rdd: RDD[(K, Array[Byte])]) {
```

Comment:
Yeah, I think how you have it, where we just add the `with` implicit class to the package object of the specific backend, makes the most sense.